website/docs/guides/examples/storybook.mdx

```diff
@@ -0,0 +1,126 @@
+---
+
+import AddDepsTabs from '@site/src/components/AddDepsTabs';
+
+Storybook is a frontend workshop for building UI components and pages in isolation. Thousands of teams use it for UI development, testing, and documentation. It’s open source and free.
+
+Storybook v7 is typically coupled with [Vite](https://vitejs.dev/). To scaffold a new Storybook project with Vite, run the following command in a project root. This guide assumes you are using React, however it is possible to use almost any (meta) framework with Storybook.
+
+```shell
+npx storybook init
+```
+
+> We highly suggest reading our documentation on [using Vite (and Vitest) with moon](./vite) 
+> and [using Jest with moon](./jest) for a more holistic view.
+
+## Setup
+
+This section assumes Storybook is being used with Vite.
+
+After setting up your Storybook, ensure your `.moon/project.yaml` has the following scripts:
+```yaml
+tasks:
+    storybook:
+        local: true
+        command: "storybook dev"
+        options:
+            outputStyle: stream
+        inputs:
+            - "@globs(sources)"
+            - ".storybook/**/*"
+```
+
+Alternatively if you only have one Storybook project, you can add the task to your `project/moon.yaml`, to run Storybook you run:
+```
+moon run <project>:storybook
+```
+
+### Vite integration
+
+Storybook 7 uses Vite out of the box, as such no configuration is required but should you choose to extend the Vite config can you do so by passing in `viteFinal`: 
+
+```ts
+// .storybook/main.js|cjs|ts
+
+const { mergeConfig } = require('vite');
+
+module.exports = {
+  stories: ['../stories/**/*.stories.mdx', '../stories/**/*.stories.@(js|jsx|ts|tsx)'],
+  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
+  core: {
+    builder: '@storybook/builder-vite',
+  },
+  async viteFinal(config) {
+    // Merge custom configuration into the default config
+    return mergeConfig(config, {
+      // Use the same "resolve" configuration as your app
+      resolve: (await import('../vite.config.js')).default.resolve,
+      // Add dependencies to pre-optimization
+      optimizeDeps: {
+        include: ['storybook-dark-mode'],
+      },
+    });
+  },
+};
+```
+
+For more information on how to integrate Vite with Storybook see the [relevant docs](https://storybook.js.org/docs/7.0/react/builders/vite#configuration).
+
+### Webpack integration
+
+If you opt to use Webpack with your Storybook instance, you can do so by passing it in the config:
+
+Be sure to install the relevant packages:
+```
+npm install --save-dev @storybook/builder-webpack5
+```
+
+```ts
+// .storybook/main.js
+
+module.exports = {
+  core: {
+    builder: '@storybook/builder-webpack5',
+  },
+};
+```
+
+For more information on how to integrate Webpack with Storybook see the [relevant docs](https://storybook.js.org/docs/7.0/react/builders/webpack).
+
+### Jest integration
+
+You can use Jest to test your stories, Storybook ships with first party plugins for improved DX
```

Comment:
"You can use Jest to test your stories, but isn't a requirement. Storybook ships with first-party plugins for improved developer experience."